packages/lab/src/number-input/NumberInput.tsx

```diff
@@ -267,9 +266,20 @@ export const NumberInput = forwardRef<HTMLDivElement, NumberInputProps>(
 
     const [displayValue, setDisplayValue] = useState<string | number>(value);
 
-    const clampAndFix = (value: number) => {
+    const clampAndFix = (value: number): number | string => {
```

Comment:
might be cleaner if only returned one type, number rather than both number | string ?

I think then you can change format to just accept numbers ?